reventlov/plugins/__init__.py

```diff
@@ -13,27 +14,77 @@ def iter_namespaces():
     )
 
 
-def find_plugins():
-    excluded = (os.getenv('REVENTLOV_DISABLED_PLUGINS') or '').split(',')
-    plugins = {
-        name.split('.')[-1]: importlib.import_module(name)
+def import_plugin_modules():
+    modules = {
+        name: importlib.import_module(name)
         for finder, name, ispkg
         in iter_namespaces()
-        if name.split('.')[-1] not in excluded
     }
-    for plugin in excluded:
-        plugins[plugin] = None
-    return plugins
-
-
-def get_plugins(dispatcher):
-    plugins = find_plugins()
-    for name in plugins:
-        if plugins[name] is not None:
-            if 'Bot' in dir(plugins[name]):
-                plugins[name] = plugins[name].Bot(dispatcher)
-            else:
-                logger.warn(
-                    f'Plugin {name} does not provide Bot class'
-                )
-    return plugins
+    return modules
+
+
+class BotPlugins(object):
+    def __init__(self, dispatcher):
+        self.dispatcher = dispatcher
+        self.modules = import_plugin_modules()
+        self.plugins = {}
+        self.load_plugins()
+
+    @property
+    def disabled_plugins(self):
+        disabled_plugins = os.getenv('REVENTLOV_DISABLED_PLUGINS')
+        if disabled_plugins is None:
+            return []
+        return disabled_plugins.split(',')
+
+    @property
+    def enabled_plugins(self):
+        return self.plugins.keys()
+
+    @property
+    def command_descs(self):
+        command_help = {}
+        for _, plugin in iteritems(self.plugins):
+            for handler in plugin.commands:
+                command = f'/{handler.command[0]}'.replace('_', '\_')
+                if handler.callback.__doc__ is None:
+                    msg = 'Undefined command'
+                else:
+                    msg = handler.callback.__doc__.splitlines()[1].strip()
+                command_help[command] = msg
+        return command_help
+
+    @property
+    def feature_descs(self):
+        return [
+            self.plugins[plugin_name].__doc__.splitlines()[1].strip()
+            for plugin_name in self.plugins
+            if self.plugins[plugin_name].__doc__ is not None
+        ]
+
+    def disable(self, plugin_name):
+        del self.plugins[plugin_name]
+        self.disabled_plugins.append(plugin_name)
+
+    def enable(self, plugin_name):
+        module_name = f'reventlov.plugins.{plugin_name}'
+        self.disabled_plugins.remove(plugin_name)
```

Comment:
This is wrong, since it doesn't really removes the plugin from the disabled_plugins.